lib/usd/ui/layerEditor/mayaCommandHook.cpp

```diff
@@ -215,30 +219,57 @@ void MayaCommandHook::lockSubLayer(UsdLayer usdLayer, MayaUsd::LayerLockType loc
 {
     std::string cmd;
     cmd = "mayaUsdLayerEditor -edit -lockLayer ";
-    // 0 = Unlocked
-    // 1 = Locked
-    // 2 = System Locked
-    switch (lockState) {
-    default:
-    case MayaUsd::LayerLockType::LayerLock_Unlocked: {
-        cmd += "0";
-        break;
-    }
-    case MayaUsd::LayerLockType::LayerLock_Locked: {
-        cmd += "1";
-        break;
-    }
-    case MayaUsd::LayerLockType::LayerLock_SystemLocked: {
-        cmd += "2";
-        break;
-    }
-    }
+    cmd += std::to_string(lockState);
 
     cmd += quote(proxyShapePath());
     cmd += quote(usdLayer->GetIdentifier());
     executeMel(cmd).asChar();
 }
 
+void MayaCommandHook::refreshLayerSystemLock(UsdLayer usdLayer, bool refreshSubLayers /*= false*/)
+{
+    if (refreshSubLayers) {
+        bool includeTopLayer = true;
+        auto allLayers = MayaUsd::getAllSublayerRefs(usdLayer, includeTopLayer);
+        for (auto layer : allLayers) {
+            _refreshLayerSystemLock(layer);
+        }
+    } else {
+        _refreshLayerSystemLock(usdLayer);
+    }
+}
+
+void MayaCommandHook::_refreshLayerSystemLock(UsdLayer usdLayer)
+{
+    // Anonymous layers do not need to be checked.
```

Comment:
This code is pretty similar to the one in the command above. IDK if it would be possible to put it in a shared function?

Reply:
They are slightly different because they are leveraging different function calls.